library/spdm_crypt_lib/fips/libspdm_selftest_rsa_pss.c

```diff
@@ -0,0 +1,173 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2023 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_crypt_lib.h"
+#include "internal/libspdm_common_lib.h"
+#include "internal/libspdm_fips_lib.h"
+#include "library/spdm_crypt_lib.h"
+
+#if LIBSPDM_FIPS_MODE
+
+/**
+ * RSA_PSS self_test
+ **/
+bool libspdm_fips_selftest_rsa_pss(void *fips_selftest_context)
+{
+    bool result = true;
+
+#if LIBSPDM_RSA_PSS_SUPPORT
+    libspdm_fips_selftest_context *context = fips_selftest_context;
+    LIBSPDM_ASSERT(fips_selftest_context != NULL);
+
+    /* any test fail cause the FIPS fail*/
+    if (context->tested_algo != context->self_test_result) {
+        return false;
+    }
+
+    /* check if run before.*/
+    if ((context->tested_algo & LIBSPDM_FIPS_SELF_TEST_RSA_PSS) != 0) {
+        return true;
+    }
+
+    uint8_t signature[LIBSPDM_MAX_ASYM_KEY_SIZE];
+    size_t sig_size;
+    void *rsa;
+
+    sig_size = sizeof(signature);
+    libspdm_zero_mem(signature, sig_size);
+
+    const uint8_t public_n[] = {
```

Comment:
please document where this is from, such as openssl.

Reply:
Thanks. I have added the comment for the test vectors.